api/base/authentication/drf.py

```diff
@@ -1,88 +1,166 @@
 import itsdangerous
+
+from django.contrib.auth import hashers
 from django.utils.translation import ugettext_lazy as _
 
 from rest_framework import authentication
 from rest_framework.authentication import BasicAuthentication
 from rest_framework import exceptions
 
+from addons.twofactor.models import UserSettings
+from api.base.exceptions import (UnconfirmedAccountError, UnclaimedAccountError, DeactivatedAccountError,
+                                 MergedAccountError, InvalidAccountError, TwoFactorRequiredError)
 from framework.auth import cas
-from framework.sessions.model import Session
-from framework.auth.core import User, get_user
+from framework.auth.core import get_user
+from osf.models import OSFUser, Session
 from website import settings
-from api.base.exceptions import UnconfirmedAccountError, DeactivatedAccountError, TwoFactorRequiredError
+
+UNUSABLE_PASSWORD_PREFIX = hashers.UNUSABLE_PASSWORD_PREFIX
 
 
 def get_session_from_cookie(cookie_val):
-    """Given a cookie value, return the `Session` object or `None`."""
+    """
+    Given a cookie value, return the `Session` object or `None`.
+
+    :param cookie_val: the cookie
+    :return: the `Session` object or None
+    """
+
     session_id = itsdangerous.Signer(settings.SECRET_KEY).unsign(cookie_val)
-    return Session.load(session_id)
+    return Session.objects.filter(_id=session_id).first()
```

Comment:
`Session.objects.get(_id=session_id)` (catch the exception if needed, `Session.ObjectDoesNotExist`)